lib/optimizely/optimizely_factory.rb

```diff
@@ -19,31 +19,46 @@
 require 'optimizely'
 module Optimizely
   class OptimizelyFactory
-    def self.create_default_instance_with_sdk_key(sdk_key)
-      Optimizely::Project.new(nil, nil, nil, nil, nil, nil, sdk_key)
+    # Returns a new optimizely instance.
+    #
+    # @params sdk_key - Required String uniquely identifying the fallback datafile corresponding to project.
+    # @param fallback - Optional JSON string datafile.
+    def self.default_instance(sdk_key, fallback = nil)
+      Optimizely::Project.new(fallback, nil, nil, nil, nil, nil, sdk_key)
     end
 
-    def self.create_default_instance_with_sdk_key_and_datafile(sdk_key, datafile = nil)
-      Optimizely::Project.new(datafile, nil, nil, nil, nil, nil, sdk_key)
-    end
-
-    def self.create_default_instance_with_config_manager(config_manager)
+    # Returns a new optimizely instance.
+    #
+    # @param config_manager - Required ConfigManagerInterface Responds to get_config.
+    def self.default_instance_with_config_manager(config_manager)
       Optimizely::Project.new(nil, nil, nil, nil, nil, nil, nil, config_manager)
     end
 
-    def self.create_default_instance(
-      datafile = nil,
+    # Returns a new optimizely instance.
+    #
+    # @params sdk_key - Required String uniquely identifying the datafile corresponding to project.
+    # @param fallback - Optional JSON string datafile.
+    # @param event_dispatcher - Optional EventDispatcherInterface Provides a dispatch_event method which if given a URL and params sends a request to it.
+    # @param logger - Optional LoggerInterface Provides a log method to log messages. By default nothing would be logged.
+    # @param error_handler - Optional ErrorHandlerInterface which provides a handle_error method to handle exceptions.
+    #                 By default all exceptions will be suppressed.
+    # @param skip_json_validation - Optional Boolean param to skip JSON schema validation of the provided datafile.
+    # @param user_profile_service - Optional UserProfileServiceInterface Provides methods to store and retreive user profiles.
+    # @param config_manager - Optional ConfigManagerInterface Responds to get_config.
+    # @param notification_center - Optional Instance of NotificationCenter.
+    def self.custom_instance(
+      sdk_key,
+      fallback = nil,
```

Comment:
Externally, this should still be called `datafile`. Internally, we can call it fallback.

Reply:
fallback is coming from java-sdk? do u still want to change it? 
https://github.com/optimizely/java-sdk/blob/master/core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java#L152

Reply:
Hmm, yeah this is a named parameter so I think we should just call it `datafile` and in the doc param we can call it the fallback datafile. I don't like the idea of introducing yet another name/concept for it.

Reply:
@mikeng13 Done